### PR TITLE
Fix the position of Toolbar while android verison >= 21

### PR DESCRIPTION
**Pull request by seasonyuu**
when the android version is Lollipop(or higher),the position of Toolbar in CheeseDetailActivity would be out of the screen....fix this

### CODE REVIEW

File: app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java [hunk 1]
```diff
@@ -42,6 +43,17 @@ public void onCreate(Bundle savedInstanceState) {
         final String cheeseName = intent.getStringExtra(EXTRA_NAME);
 
         final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        if(Build.VERSION.SDK_INT>=21){
```
Comment (steffandroid) [inline app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java]: Change to `Build.VERSION.SDK_INT == 21` as this issue is only present at that API level.

File: app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java [hunk 2]
```diff
@@ -42,6 +43,17 @@ public void onCreate(Bundle savedInstanceState) {
...
+            int result = 0;
+            int resourceId = getResources().getIdentifier(
+                    "status_bar_height", "dimen", "android");
+            if (resourceId > 0) {
+                result = getResources().getDimensionPixelSize(resourceId);
```
Comment (steffandroid) [inline app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java]: Multiply result by 2.

File: app/src/main/res/layout/activity_detail.xml [hunk 3]
```diff
@@ -51,6 +51,7 @@
                 android:id="@+id/toolbar"
                 android:layout_width="match_parent"
                 android:layout_height="?attr/actionBarSize"
+                android:fitsSystemWindows="true"
```
Comment (steffandroid) [inline app/src/main/res/layout/activity_detail.xml]: Not necessary with aforementioned alterations.

Comment (seasonyuu) [inline app/src/main/res/layout/activity_detail.xml]: @steffandroid really??I had removed this line , but performs badly....do you have a try?

File: app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java (hunk 2 above)
Comment (seasonyuu) [inline app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java]: @steffandroid what do u mean?

File: app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java (hunk 1 above)
Comment (seasonyuu) [inline app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java]: I have no other device to test , my mistake

File: app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java (hunk 2 above)
Comment (steffandroid) [inline app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java]: Sorry, I should have explained this better. If you make the `Build.VERSION.SDK_INT == 21` alteration above, you also need to remove `android:fitsSystemWindows="true"` below as it causes the Toolbar to be positioned incorrectly on API level 22. Once those changes are made, you also need to change the above line to `result = getResources().getDimensionPixelSize(resourceId) * 2;` to position the Toolbar correctly on API level 21.

Comment (seasonyuu) [inline app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java]: I found that you are right on the problem of android version , but I still
don't understand why the result should be double.I run what I wrote in my
phone(Xiaomi aries android 5.0.2), and the position of Toolbar seems no
problem ......

2015-06-09 18:12 GMT+08:00 Steffan Davies <notifications@github.com>:

File: app/src/main/res/layout/activity_detail.xml (hunk 3 above)
Comment (rnascimento13) [inline app/src/main/res/layout/activity_detail.xml]: private void fixApi21ToolBarBug(Toolbar toolbar){ //bug that puts toolbar out of screen
        if(Build.VERSION.SDK_INT!=21) return; //only on api 21
        final int resourceId = getResources().getIdentifier("status_bar_height", "dimen", "android");
        final int result = (resourceId>0) ? getResources().getDimensionPixelSize(resourceId)*2 : 0;
        final CollapsingToolbarLayout.LayoutParams params = 
                (CollapsingToolbarLayout.LayoutParams)toolbar.getLayoutParams();
        params.topMargin -= result;
        toolbar.setLayoutParams(params);
    }

**Comment by rnascimento13** [comment]
this worked for me

    private void fixApi21ToolBarBug(Toolbar toolbar){ //bug that puts toolbar out of screen
        if(Build.VERSION.SDK_INT!=21) return; //only on api 21
        final int resourceId = getResources().getIdentifier("status_bar_height", "dimen", "android");
        final int result = (resourceId>0) ? getResources().getDimensionPixelSize(resourceId)*2 : 0;
        final CollapsingToolbarLayout.LayoutParams params = 
                (CollapsingToolbarLayout.LayoutParams)toolbar.getLayoutParams();
        params.topMargin -= result;
        toolbar.setLayoutParams(params);
    }

**Comment by rnascimento13** [comment]
this worked for me

    //bug that puts toolbar out of screen
    private void fixApi21ToolBarBug(Toolbar toolbar) {
        if(Build.VERSION.SDK_INT!=21) return; //only on api 21
            final int resourceId = getResources().getIdentifier("status_bar_height", "dimen", "android");
            final int result = (resourceId>0) ? getResources().getDimensionPixelSize(resourceId)*2 : 0;
            final CollapsingToolbarLayout.LayoutParams params = 
                (CollapsingToolbarLayout.LayoutParams)toolbar.getLayoutParams();
            params.topMargin -= result;
            toolbar.setLayoutParams(params);
    }

**Pull request closed without merging**